apps/api/src/database/repositories/competition-repository.ts

```diff
@@ -929,7 +948,13 @@ async function getAgentPortfolioSnapshotsImpl(
         ),
       )
       .orderBy(desc(portfolioSnapshots.timestamp));
-    // TODO: there's no limit here, this is a weakness in perf
+
+    // Apply limit if specified
+    if (limit !== undefined && limit > 0) {
```

Comment:
when an agent requests its portfolio, we just get the most recent snapshot (limit 1)